### ISSUE TITLE
PreferJavaTimeOverload generates false positive with jOOQ DSL.inline()

### ISSUE THREAD

**Issue by tbroyer** [google/error-prone#1504]
### Description of the problem:

when using jOOQ's `DSL.inline(1)`, ErrorProne emits a warning (which fails our build because we use `-Werror`)
```
warning: [PreferJavaTimeOverload] If the numeric primitive (1) represents a Instant, please call DSL.inline(Instant) instead.
```

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Simply call `DSL.inline(1)`.

### What version of Error Prone are you using?

2.3.4

###  Have you found anything relevant by searching the web?

PreferJavaTimeOverload seems to check the method return type when looking for overloads, but in this case [`DSL.inline(int)`](https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/impl/DSL.html#inline(int)) returns a `Param<Integer>` while [`DSL.inline(Instant)`](https://www.jooq.org/javadoc/latest/org.jooq/org/jooq/impl/DSL.html#inline(java.time.Instant)) returns a `Param<Instant>`.
IMO, they should not be treated as the same return type.
